### ISSUE TITLE
explTF and expstar destinations do not exist

### ISSUE THREAD

**Issue by blefloch** [latex3/latex3#275]
When documenting any conditional or expandable function, a hyperlink is made from `TF` or from a star to a non-existing destination (only existing when expl3 is typeset).  This leads to

```
pdfTeX warning (dest): name{explTF} has been referenced but does not exist, replaced by a fixed one
pdfTeX warning (dest): name{expstar} has been referenced but does not exist, replaced by a fixed one
```

Presumably fixing that requires keeping track in the `aux` file whether the destinations are defined anywhere.  I have some recollection that there is a package doing precisely that (maybe by Heiko?).

**Comment by wspr** [comment]
I couldn't find an Oberdiek package to do this unfortunately. Two thoughts here: 

1. Patch `\hypertarget` to write data to the aux file. Not a bad approach, really — when cross-referencing I like being able to detect if a LaTeX reference exists, and doing something similar with hyperref would be natural.

2. Or provide special `\CodedocHypertarget` commands just for this purpose, which does a more lightweight version of setting a flag whether to hyperlink these symbols or not.

Right now I'm leaning towards option 1. Regardless, maybe time to revive the `auxdata` work I was doing some time back.

**Comment by blefloch** [comment]
Fixed by the same commit as #477 since they were actually duplicates.
